bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1790,6 +1851,14 @@ void handleDelayedWriteBookieFailure() {
             }
             return;
         }
+        if (writeFlags.contains(WriteFlag.DEFERRED_SYNC)) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Cannot perform ensemble changee with writeflags {}."
+                        + "Failed bookies {} for ledger {}.",
+                        writeFlags, delayedWriteFailedBookies, ledgerId);
+            }
```

Comment:
Maybe we should treat this also unrecoverable error, or even fail at the caller itself.
Because if AQ< WQ we can continue to write but force is going to fail anyway. 
To be consistent, any write error (even if it is delayed write error it needs to fail under this flag)

Reply:
actually with writeflag = DEFERRED_SYNC this block of code will never run.
I have copied this block for consistency with other cases of "enabled" disableEnsembleChangeFeature.

I can drop it if you prefer, this code will never run